test/src/test/resources/fixtures/hydration/conditional-hydration-equals-predicate-works-with-integer-type.yml

```diff
@@ -0,0 +1,161 @@
+name: "conditional hydration equals predicate works with integer type"
+enabled: true
+overallSchema:
+  service2: |
+    type Query {
+      barById(id: ID): Bar
+    }
+    type Bar {
+      id: ID
+      name: String
+      type: Int
+    }
+
+  service1: |
+    type Query {
+      foo: Foo
+    }
+    type Foo {
+      id: ID
+      type: Int
+      bar: Bars
+      @hydrated(
+        service: "service2"
+        field: "barById"
+        arguments: [
+          {
+            name: "id" 
+            value: "BAR_A"
+          }
+        ]
+        when: {
+          result: {
+              sourceField: "type"
+              predicate: { equals: 1 }
+          }
+        }
+      )
+      @hydrated(
+        service: "service2" 
+        field: "barById" 
+        arguments: [
+          {
+            name: "id" 
+            value: "BAR_B"
+          }
+        ]
+        when: {
+          result: {
+              sourceField: "type"
+              predicate: { equals: 2 }
+          }
+        }
+      )
+    }
+    union Bars = Bar
+underlyingSchema:
+  service2: |
+    type Bar {
+      id: ID
+      name: String
+      type: Int
+    }
+    union Bars = Bar
+    
+    type Query {
+      barById(id: ID): Bar
+    }
+  service1: |
+    type Foo {
+      barId: ID
+      id: ID
+      type: Int
+    }
+    
+    type Query {
+      foo: Foo
+    }
+query: |
+  query {
+    foo {
+      bar {
+        ... on Bar {
+          name
+        }
+      }
+    }
+  }
+variables: { }
+serviceCalls:
+  - serviceName: "service1"
+    request:
+      query: |
+        query {
+          foo {
+            __typename__hydration__bar: __typename
+            hydration__bar__type: type
+            hydration__bar__type: type
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "foo": {
+            "__typename__hydration__bar": "Foo",
+            "hydration__bar__type": 2
+          }
+        },
+        "extensions": {}
+      }
+  - serviceName: "service2"
+    request:
+      query: |
+        query {
+          barById(id: "BAR_A") {
+            name
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "barById": {
+            "name": "Bar A"
+          }
+        },
+        "extensions": {}
+      }
```

Comment:
This is NOT used, so remove it. The test framework should really error out if a service call isn't used.

```suggestion
```